openssl-dynamic/src/test/java/io/netty/internal/tcnative/CertificateVerifierTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.internal.tcnative;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+
+public class CertificateVerifierTest extends AbstractNativeTest {
+
+    @Test
+    public void testValidErrorCode() throws Exception {
+        Field[] fields = CertificateVerifier.class.getFields();
+        for (Field field : fields) {
+            if (field.isAccessible()) {
+                int errorCode = field.getInt(null);
+                Assert.assertTrue(CertificateVerifier.isValid(errorCode));
+            }
+        }
+    }
+
+    @Test
+    public void testNonValidErrorCode() {
+        // Will overflow on first increment
+        int errorCode = Integer.MAX_VALUE;
+        do {
+            errorCode++;
+            Assert.assertTrue(errorCode < Integer.MAX_VALUE);
+        } while (CertificateVerifier.isValid(errorCode));
```

Comment:
What is the intention of this test? To prove that at least 1 negative value is invalid? Seems strange in general. If we are just trying to demonstrate there is at least 1 error code that is not valid we have a few options:

- create some private markers to indicate "highest" and "lowest" valid value and use these in the test. Negative of this approach is we have to worry about updating these just for testing.
- make the set package private, sort all the values, and pick a value not in the set.
- randomly choose values until we find one which is not in the set and make sure its not valid.